### DISCUSSION TITLE
update specific key's value in jsonb[] column using rpc function

### DISCUSSION THREAD

**Discussion by amrajat** [supabase/supabase#23096] [Questions]
Greeting! 

I want a function that asks the id[takes parmater as tune_id] to match the value of the "id" key in any object in array and update its "downloaded" key's value from false to true.  it updates only the downloaded value to true for the object with the specified id. in this case the id can be either 1234289 or 1593852. value of "id" key is type integer.

Data looks like this in my table editor of studios column in my users' table defied as jsonb[] Array

![image](https://github.com/supabase/supabase/assets/26631696/1f7d297d-4aa1-453a-973d-2f7aebcb93f1)

```
[
  {
    "id": 1234289,
    "downloaded": false
  },
  {
    "id": 1593852,
    "downloaded": false
  }
]
```

The Function im using is below:
```
CREATE OR REPLACE FUNCTION update_downloaded_value(tune_id integer, user_id text)
RETURNS void AS $$
BEGIN
  UPDATE users
  SET studios = jsonb_set(studios, concat((SELECT index FROM jsonb_array_elements(studios) WITH ORDINALITY WHERE (value->>'id')::integer = tune_id), '/downloaded'), 'true')
  WHERE id = user_id::uuid;
END;
$$ LANGUAGE plpgsql;
```
The Error:
```
{
  code: '42883',
  details: null,
  hint: 'No function matches the given name and argument types. You might need to add explicit type casts.',
  message: 'function jsonb_array_elements(jsonb[]) does not exist'
}
```
Thanks!

**Comment by GaryAustin1** [comment]
I don't have the way to do what you want, but you are treating the column like a json column in your code.   You could have a json array inside of a json column maybe to do that.  But you have an array column that each element is json.  The code you are running (jsonb_array_elements) is looking for a json array not an array of json.
